pkg/workflows/internal/v2/sdkimpl/runtime_test.go

```diff
@@ -0,0 +1,332 @@
+	actionandtriggermock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/actionandtrigger/action_and_triggermock"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basicaction"
+	basicactionmock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basicaction/basic_actionmock"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basictrigger"
+	basictriggermock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/basictrigger/basic_triggermock"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/nodeaction"
+	nodeactionmock "github.com/smartcontractkit/chainlink-common/pkg/capabilities/v2/protoc/pkg/test_capabilities/nodeaction/node_actionmock"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/testutils"
+)
+
+var anyTrigger = &basictrigger.Outputs{CoolOutput: "cool"}
+var anyConfig = &basictrigger.Config{Name: "name", Number: 123}
+
+func TestRuntime_CallCapability(t *testing.T) {
+	t.Run("runs async", func(t *testing.T) {
+		ch := make(chan struct{}, 1)
+		anyResult1 := "ok1"
+		action1, err := basicactionmock.NewBasicActionCapability(t)
+		require.NoError(t, err)
+		action1.PerformAction = func(_ context.Context, input *basicaction.Inputs) (*basicaction.Outputs, error) {
+			<-ch
+			return &basicaction.Outputs{AdaptedThing: anyResult1}, nil
+		}
+
+		anyResult2 := "ok2"
+		action2, err := actionandtriggermock.NewBasicCapability(t)
+		action2.Action = func(ctx context.Context, input *actionandtrigger.Input) (*actionandtrigger.Output, error) {
+			return &actionandtrigger.Output{Welcome: anyResult2}, nil
+		}
+
+		test := func(rt sdk.DonRuntime, _ *basictrigger.Outputs) (string, error) {
+			workflowAction1 := &basicaction.BasicAction{}
+			call1 := workflowAction1.PerformAction(rt, &basicaction.Inputs{InputThing: true})
+
+			workflowAction2 := &actionandtrigger.Basic{}
+			call2 := workflowAction2.Action(rt, &actionandtrigger.Input{Name: "input"})
+			result2, err := call2.Await()
+			require.NoError(t, err)
+			ch <- struct{}{}
+			result1, err := call1.Await()
+			require.NoError(t, err)
+			return result1.AdaptedThing + result2.Welcome, nil
+		}
+
+		ran, result, err := testRuntime(t, test)
+		require.NoError(t, err)
+		assert.True(t, ran)
+		assert.Equal(t, anyResult1+anyResult2, result)
+	})
+
+	t.Run("call capability errors", func(t *testing.T) {
+		test := func(rt sdk.DonRuntime, _ *basictrigger.Outputs) (string, error) {
+			workflowAction1 := &basicaction.BasicAction{}
+			call := workflowAction1.PerformAction(rt, &basicaction.Inputs{InputThing: true})
+			_, err := call.Await()
```

Comment:
why does this error?

Reply:
The capability isn't in the registry, so call capability returns an error. I'm testing that callback errors are handled correctly.

Reply:
I added a comment